### PR TITLE
Murad

### PR DESCRIPTION
**Pull request by Hdvndgr**
Javascript2 Week1 Homework.

### CODE REVIEW

File: Week1/homework/app.js [hunk 1]
```diff
@@ -1,11 +1,165 @@
+/* eslint-disable camelcase */
+
 'use strict';
 
 {
+  // #region 1.1_&_1.2
   const bookTitles = [
-    // Replace with your own book titles
-    'harry_potter_chamber_secrets',
+    'The 7 Habits of Highly Effective People',
```
Comment (wilgert) [inline Week1/homework/app.js]: I know that the instructions and code sample were a bit confusing. The idea was to directly create a list of bookIds here. Just like the example: `harry_potter_chamber_secrets`.

File: Week1/homework/app.js [hunk 2]
```diff
@@ -1,11 +1,165 @@
...
+  /*
+  bookTitles.forEach(item => {
+    console.log(item);
+  });
+  */
+  // #endregion 1.1_&_1.2
+
+  // #region 1.3
+  function idNameConvention(name) {
+    const reg = /\w+/gi;
```
Comment (wilgert) [inline Week1/homework/app.js]: I do like the creative use of regex to turn a normal title into a valid id!

File: Week1/homework/app.js [hunk 3]
```diff
@@ -1,11 +1,165 @@
...
+    const myString = name.replace(reg, function(s) {
+      return s.toLowerCase();
+    });
+    return myString.replace(/ /gi, '_');
+  }
+
+  function generate_List() {
```
Comment (wilgert) [inline Week1/homework/app.js]: Why did you break the JavaScript naming convention here? Function names should be camelCase.

File: Week1/homework/app.js [hunk 4]
```diff
@@ -1,11 +1,165 @@
...
+    const ul = document.createElement('ul');
+    bookTitles.forEach(item => {
```
Comment (wilgert) [inline Week1/homework/app.js]: Nice `forEach` loop!

File: Week1/homework/app.js [hunk 5]
```diff
@@ -1,11 +1,165 @@
...
+    });
+
+    document.body.appendChild(ul);
+  }
+
+  generate_List();
+  // #endregion 1.3
+
+  // #region 1.4
+  const bookDetails = ['Title', 'Language', 'Author'];
```
Comment (wilgert) [inline Week1/homework/app.js]: Again the assignment might have been unclear but the idea was to directly create an object like this:
```
const bookObject = {
	'harry_potter_chamber_secrets': {
		title: 'Harry Potter and the Chamber of Secrets',
		author: 'J.K. Rowling',
		language: 'English'
	}
}
```

File: Week1/homework/app.js [hunk 6]
```diff
@@ -1,11 +1,165 @@
...
+    }
+  };
+
+  function addBook(bookList, book) {
+    bookList[idNameConvention(book.title) + '-2'] = book;
+  }
+
+  function createLibrary() {
+    const library = {};
+    const book_Infos = [
```
Comment (wilgert) [inline Week1/homework/app.js]: Also names of variables should be camelCase.

File: Week1/homework/app.js [hunk 7]
```diff
@@ -1,11 +1,165 @@
...
+      [bookTitles[7], 'English', 'Paulo Coelho'],
+      [bookTitles[8], 'English', 'Sun Tzu'],
+      [bookTitles[9], 'Turkish', 'Mehmet Ali Bulut'],
+    ];
+
+    for (let index = 0; index < book_Infos.length; index++) {
+      const b = new Book(bookDetails, book_Infos[index]);
+      addBook(library, b);
+    }
+    return library;
```
Comment (wilgert) [inline Week1/homework/app.js]: I now see that this method creates an Object in the shape that was asked for. Although it is a creative way of doing it, it is also a lot more complex for no benefit.

File: Week1/homework/app.js [hunk 8]
```diff
@@ -1,11 +1,165 @@
...
+      li.id = key;
+      li.classList.add('book_item');
+      li.style = 'clear:both; margin: 80px 0;';
+      const ul_book = document.createElement('ul');
+      ul_book.classList.add('details');
+      for (const [key_book, value_book] of Object.entries(value)) {
+        const li_book = document.createElement('li');
+        const heading = document.createElement('h3');
+        const sp = document.createElement('span');
+        sp.innerText = key_book.charAt(0).toUpperCase() + key_book.slice(1) + ': ';
```
Comment (wilgert) [inline Week1/homework/app.js]: Nice way of turning the object key in to a capitalized string.

File: Week1/homework/app.js [hunk 9]
```diff
@@ -1,11 +1,165 @@
...
...PAzk3b8Pyh25Y01C9pfKjBs/1HXJ+OggrOCLxqZJdT1utQEVilEZOj70BnvfpUP7LH+T7jVuE0J96ZGtg48sixOKIEzD7j4Z82sDdgagLfPox0Bai6CFFIw0bzb3O1c997bUVC24DqfFFQ6xnxL2dt1zLo7CIs4z1ogJ/Kgk4m8uRfc+uuqdd10uhe/k3532NSj9g8jEl055xOemXlMsSbF7ApncW1OOrWxczxFAKUKN/xMtftbSTuWLmH7jiWte3VAADCLD8+Pk/b6Cs4uPIxvNDByHNRhR7P+EwuTPRpI1uMjuoTDL8F9v95GbefVZnJJpBmbgpy5wyUu7UZyK9xEQE2Zqp/sTYl8rC7RNTV413INljb7SBSA3ex2CCCw2ncNWOQV5ZS0btW2FzTQA3MOaiOYwTKlQ/Umyl+QypEIXVucHQy8MRcKbPuD4wbFYB3mXTuEOSVIMdOCiAdSozFvboVSpi95p00bU6skZb2XFdpTH5F7w9e1dYwk5tKlsN7+pljUk5VM2h5zeENAu6CgJwKfY4muk6+Er3ILDb40myql5+sT6oBxPmSM/63Bd9d32HL9MptE4ij8HqtyeqHSmksPn87uL2j6cV1TxlXjRofE10c9WAVnRrTAl16LTBha7jIMBnzXdrvtTmyminT+qVRRHlSGubDvE7qYkngO9e/dOiYYc+Jai9kh3c1XcD3BwjvY4NC54OeFP5yX726pIpRxG+FbV/GlaHc2zGJuYfCI159xu71Fz98beq6bDVXio7J3i50hFHkAJLqOkEmLY2YH2aS1kRx4f6ADPcOuQgJ3FCDCtNkGzMUnw8ZlBE2jcNjix+/DInlCNmAEpClmNhS/0GwsaapVy/3U1dE4xeI0SAPKqaAFY7/bkctt4W5VUddv3KsvBsDpU80raJEmyuZ4LcmtHKJt42zUebWOobCcERtAR0aHt6tNcj2+fJdA4LJ937hMTuCqYdQIGJwFu3FizuQbebu4KR2tWLy7jJ8AkIiM/hllpn28FI02uTGu6GH0F8wqiSqmGR+EC4WyYV9frm/TB69qYgY/36ndC1X/gkdmHc4pop0K5uObBZJKe9ErQIKCM4isTDwARQYEn9DEOBHJCJ9yBkx0AZrSmAA9A+adcgANACuIpRyyRSVuqoBHapWmuFvhHgloMZPL4vOfcZeF5CBlMjrlCoAAAA=',
```
Comment (wilgert) [inline Week1/homework/app.js]: I would not put large images as base64 into js like this. It makes the js file very big. Only do this with small images like icons. Store larger images as separate files.

**Comment by fedefu** [comment]
@Hdvndgr Great job on the homework quality and speed, but can you please reply and implement feedback? Thanks!

File: Week1/homework/app.js (hunk 2 above)
Comment (Hdvndgr) [reply Week1/homework/app.js]: Thanks.

File: Week1/homework/app.js (hunk 3 above)
Comment (Hdvndgr) [reply Week1/homework/app.js]: I found it much more readable. Next time I will use camelCase.

File: Week1/homework/app.js (hunk 6 above)
Comment (Hdvndgr) [reply Week1/homework/app.js]: Ok. I will show more attention.

File: Week1/homework/app.js (hunk 7 above)
Comment (Hdvndgr) [reply Week1/homework/app.js]: Right. I just confused and couldn't think easier.

**Pull request closed without merging**